From bc7c22610b0ae040d8f465a258dc4d578d4919df Mon Sep 17 00:00:00 2001
From: David Pacheco <dap@joyent.com>
Date: Mon, 12 Sep 2016 15:28:28 -0700
Subject: [PATCH] joyent/node-manta#261 "AssertionError: undefined (object) is
 required" after "socket hang up" Reviewed by: Cody Peter Mello
 <cody.mello@joyent.com> Reviewed by: Brian Bennett <brian.bennett@joyent.com>
 Approved by: Cody Peter Mello <cody.mello@joyent.com>

---
 CHANGES.md    | 5 +++++
 lib/client.js | 2 +-
 package.json  | 2 +-
 3 files changed, 7 insertions(+), 2 deletions(-)

diff --git a/CHANGES.md b/CHANGES.md
index d9fb049..b0ed749 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -2,6 +2,11 @@
 
 ## not yet released
 
+## 3.1.1
+
+- joyent/node-manta#261 "AssertionError: undefined (object) is required" after
+  "socket hang up"
+
 ## 3.1.0
 
 - joyent/node-manta#265 Add `--json, -j` option to `mfind`. E.g.:
diff --git a/lib/client.js b/lib/client.js
index 86407ba..f1e4485 100644
--- a/lib/client.js
+++ b/lib/client.js
@@ -384,7 +384,7 @@ function onResultLineStreamCallback(opts) {
 
 function readError(err, res, cb) {
     assert.object(err);
-    assert.object(res);
+    assert.optionalObject(res);
     assert.func(cb);
 
     if (res === null)
diff --git a/package.json b/package.json
index 0219f86..e3e7c4c 100644
--- a/package.json
+++ b/package.json
@@ -7,7 +7,7 @@
         "type": "git",
         "url": "git://github.com/joyent/node-manta.git"
     },
-    "version": "3.1.0",
+    "version": "3.1.1",
     "main": "./lib/index.js",
     "dependencies": {
         "assert-plus": "^1.0.0",
-- 
2.21.0

